tests/subtractions/test_api.py

```diff
@@ -144,268 +112,344 @@ async def test_get_from_job(fake: DataFaker, spawn_job_client, snapshot_recent):
         authenticated=True,
         permissions=[Permission.modify_subtraction],
     )
 
-    await mongo.subtraction.insert_one(document)
-
-    resp = await client.patch("/subtractions/apple", data)
+    resp = await client.patch(f"/subtractions/{subtraction.id}", data)
 
-    assert resp.status == 200
-    assert await resp.json() == snapshot
-    assert await mongo.subtraction.find_one() == snapshot
+    assert resp.status == HTTPStatus.OK
+    assert await resp.json() == snapshot_recent
+    assert await mongo.subtraction.find_one() == snapshot_recent
 
 
 @pytest.mark.parametrize("exists", [True, False])
-async def test_delete(
+async def test_remove_by_user(
     exists: bool,
-        fake: DataFaker,
-    resp_is,
-    mongo: Mongo,
+    fake: DataFaker,
     spawn_client: ClientSpawner,
-    tmp_path,
 ):
     client = await spawn_client(
         authenticated=True,
         permissions=[Permission.modify_subtraction],
     )
 
-    get_config_from_app(client.app).data_path = tmp_path
-
     if exists:
         user = await fake.users.create()
-        job = await fake.jobs.create(user)
-
-        await mongo.subtraction.insert_one(
-            {
-                "_id": "foo",
-                "name": "Foo",
-                "deleted": False,
-                "ready": False,
-                "nickname": "Foo Subtraction",
-                "user": {"id": user.id},
-                "job": {"id": job.id},
-            },
+        upload = await fake.uploads.create(
+            user=user,
+            upload_type=UploadType.subtraction,
+            name="foobar.fq.gz",
+        )
+        subtraction = await fake.subtractions.create(
+            user=user,
+            upload=upload,
+            finalized=True,
         )
 
-    resp = await client.delete("subtractions/foo")
+        resp = await client.delete(f"subtractions/{subtraction.id}")
 
-    assert resp.status == 204 if exists else 400
+        assert resp.status == HTTPStatus.NO_CONTENT
+    else:
+        resp = await client.delete("subtractions/NOT_FOUND_ID")
+        assert resp.status == HTTPStatus.NOT_FOUND
 
 
-@pytest.mark.parametrize("error", [None, "404_name", "404", "409"])
-async def test_upload(
-    error: str | None,
-    pg: AsyncEngine,
-    resp_is,
-    mongo: Mongo,
-    spawn_job_client,
-    snapshot,
-    tmp_path: Path,
-):
-    client = await spawn_job_client(authenticated=True)
-    test_dir = tmp_path / "files"
-    test_dir.mkdir()
-    test_dir.joinpath("subtraction.1.bt2").write_text("Bowtie2 file")
+class TestUploadSubtractionFile:
+    @pytest.fixture(autouse=True)
+    async def setup(
+        self,
+        fake: DataFaker,
+        tmp_path: Path,
+    ):
+        test_dir = tmp_path / "files"
+        test_dir.mkdir(exist_ok=True)
 
-    get_config_from_app(client.app).data_path = tmp_path
+        self.test_file_directory: Path = test_dir
+        self.test_subtraction_file_directory: Path = tmp_path / "subtractions"
```

Comment:
I don't think this should be necessary, unless `data_path` is not working as expected.

Reply:
Might need to call to clarify this one.